<a href="https://colab.research.google.com/github/RobertoCalber/DataScienceSeminar/blob/main/TallerCorrelacionPearson.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import pandas as pd
import numpy as np

# Cargar archivo
ruta = "/content/clientes_dataset.xlsx"   # en Colab sube aquí el archivo
df = pd.read_excel(ruta, sheet_name="Hoja1")

print("Dimensión del dataset:", df.shape)
print("\nColumnas:")
print(df.columns.tolist())
print("\nTipos de datos:")
print(df.dtypes)
print("\nPrimeras filas:")
display(df.head())

Dimensión del dataset: (10, 10)

Columnas:
['edad', 'ingreso_mensual', 'visitas_web_mes', 'tiempo_sitio_min', 'paginas_vistas', 'clics_email', 'compras_previas', 'uso_descuento', 'satisfaccion', 'abandono_cliente']

Tipos de datos:
edad                int64
ingreso_mensual     int64
visitas_web_mes     int64
tiempo_sitio_min    int64
paginas_vistas      int64
clics_email         int64
compras_previas     int64
uso_descuento       int64
satisfaccion        int64
abandono_cliente    int64
dtype: object

Primeras filas:


,edad,ingreso_mensual,visitas_web_mes,tiempo_sitio_min,paginas_vistas,clics_email,compras_previas,uso_descuento,satisfaccion,abandono_cliente
0,25,1800,12,7,14,2,1,1,3,1
1,34,3200,5,6,9,0,3,0,4,0
2,29,2100,9,8,12,1,2,1,3,1
3,45,4000,4,5,8,0,4,0,5,0
4,31,2600,10,9,15,3,2,1,4,0


**¿Cuántas variables tiene el dataset?**

10 variables

**¿Cuál es la variable objetivo?**

abandono_cliente

**¿Qué tipo de variables observas?**

Todas son numéricas

**Paso 3 — Identificar correlación con la variable objetivo**

In [2]:
# Correlación de Pearson
corr = df.corr(numeric_only=True)

print("Matriz de correlación:")
display(corr.round(4))

Matriz de correlación:


,edad,ingreso_mensual,visitas_web_mes,tiempo_sitio_min,paginas_vistas,clics_email,compras_previas,uso_descuento,satisfaccion,abandono_cliente
edad,1.0000,0.9857,-0.9382,-0.8168,-0.8863,-0.7916,0.9570,-0.8360,0.9546,-0.7945
ingreso_mensual,0.9857,1.0000,-0.9571,-0.8210,-0.8928,-0.7913,0.9672,-0.8865,0.9616,-0.8414
visitas_web_mes,-0.9382,-0.9571,1.0000,0.8771,0.9643,0.9077,-0.9584,0.8736,-0.9031,0.7791
tiempo_sitio_min,-0.8168,-0.8210,0.8771,1.0000,0.9402,0.9251,-0.8170,0.7707,-0.7707,0.5581
paginas_vistas,-0.8863,-0.8928,0.9643,0.9402,1.0000,0.9729,-0.9057,0.8098,-0.8414,0.6453
clics_email,-0.7916,-0.7913,0.9077,0.9251,0.9729,1.0000,-0.8116,0.7332,-0.7332,0.5419
compras_previas,0.9570,0.9672,-0.9584,-0.8170,-0.9057,-0.8116,1.0000,-0.8807,0.8807,-0.7206
uso_descuento,-0.8360,-0.8865,0.8736,0.7707,0.8098,0.7332,-0.8807,1.0000,-0.7917,0.6667
satisfaccion,0.9546,0.9616,-0.9031,-0.7707,-0.8414,-0.7332,0.8807,-0.7917,1.0000,-0.8750
abandono_cliente,-0.7945,-0.8414,0.7791,0.5581,0.6453,0.5419,-0.7206,0.6667,-0.8750,1.0000


In [3]:
# Variable objetivo
objetivo = "abandono_cliente"

# Variables que aparecen en la tabla
variables_tabla = [
    "edad",
    "ingreso_mensual",
    "visitas_web_mes",
    "tiempo_sitio_min",
    "paginas_vistas",
    "clics_email",
    "compras_previas",
    "uso_descuento",
    "satisfaccion"
]

tabla_taller = pd.DataFrame({
    "Variable": variables_tabla,
    "Correlación con abandono_cliente": [corr.loc[var, objetivo] for var in variables_tabla],
    "Tipo de relación (positiva/negativa)": [
        "positiva" if corr.loc[var, objetivo] > 0 else "negativa"
        for var in variables_tabla
    ]
})

tabla_taller["Correlación con abandono_cliente"] = tabla_taller["Correlación con abandono_cliente"].round(4)
display(tabla_taller)

,Variable,Correlación con abandono_cliente,Tipo de relación (positiva/negativa)
0,edad,-0.7945,negativa
1,ingreso_mensual,-0.8414,negativa
2,visitas_web_mes,0.7791,positiva
3,tiempo_sitio_min,0.5581,positiva
4,paginas_vistas,0.6453,positiva
5,clics_email,0.5419,positiva
6,compras_previas,-0.7206,negativa
7,uso_descuento,0.6667,positiva
8,satisfaccion,-0.8750,negativa


**Paso 4 — Selección de variables más relevantes**

**¿Cuáles son las 3 variables con mayor correlación en valor absoluto?**

satisfaccion → -0.8750

ingreso_mensual → -0.8414

edad → -0.7945

**¿Qué variable tiene la correlación más alta?**

satisfaccion


**Paso 5 — Identificación de variables redundantes**


edad — ingreso_mensual — 0.9857 — Sí

paginas_vistas — clics_email — 0.9729 — Sí

visitas_web_mes — paginas_vistas — 0.9643 — Sí

visitas_web_mes — compras_previas — -0.9584 — Sí


**Paso 6 — Construcción del dataset reducido**

satisfaccion

ingreso_mensual

visitas_web_mes

uso_descuento

tiempo_sitio_min



**1. ¿Qué significa una correlación negativa en este contexto?**

Una correlación negativa significa que cuando una variable aumenta, la variable abandono_cliente tiende a disminuir. En este contexto, indica una relación inversa con el abandono. Por ejemplo, si aumenta la satisfacción o el ingreso mensual, la tendencia al abandono del cliente disminuye.

**2. Explica la relación entre satisfaccion y abandono_cliente.**

La relación entre satisfaccion y abandono_cliente es negativa y fuerte, con un valor de -0.8750. Esto indica que, a mayor satisfacción del cliente, menor es la probabilidad de abandono. En otras palabras, los clientes más satisfechos tienden a permanecer, mientras que los menos satisfechos presentan mayor tendencia a abandonar.

**3. Explica por qué una variable con correlación cercana a 0 no es útil.**

Una variable con correlación cercana a 0 no es muy útil en este análisis porque no muestra una relación lineal clara con la variable objetivo. Eso significa que su cambio no ayuda a explicar de forma directa el comportamiento de abandono_cliente, por lo que aporta poca información para la selección de atributos usando correlación de Pearson.

**4. ¿Por qué no debemos incluir variables altamente correlacionadas entre sí?**

No se deben incluir variables altamente correlacionadas entre sí porque contienen información muy parecida o redundante. Esto no mejora el análisis y puede hacer que el modelo sea más complejo sin aportar valor adicional. Por eso, en reducción de dimensionalidad conviene quedarse solo con una de las variables redundantes o con la más relevante frente a la variable objetivo.